### PR TITLE
Support Go Modules

### PR DESCRIPTION
**Pull request by hayajo**
Support [Go Modules](https://github.com/golang/go/wiki/Modules).

- add go.{mod,sum}
- add 'lint-deps' and 'update' task

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,39 +2,36 @@ BIN := mackerel-container-agent
...
-lint: test-deps
+lint: lint-deps
 	go vet ./...
 	golint -set_exit_status ./...
 
 .PHONY: clean
 clean:
 	rm -fr build
-	go clean
+	GO111MODULE=off go clean
```
Comment (itchyny) [inline Makefile]: Why are you disabling go module here?

Comment (hayajo) [reply Makefile]: In the case of GO111MODULE = on, the following error will occur.

```
$ go clean
can't load package: package github.com/mackerelio/mackerel-container-agent: unknown import path "github.com/mackerelio/mackerel-container-agent": cannot find module providing package github.com/mackerelio/mackerel-container-agent
```

Comment (itchyny) [reply Makefile]: Ah, try `go clean ./...`.

Comment (hayajo) [reply Makefile]: Thank you!

**Comment by itchyny** [comment]
LGTM!

**Comment by hayajo** [comment]
Thank you!

**Pull request merged**